pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -444,21 +444,49 @@ func (pc *PodController) configurePodSecondaryNetwork(pod *corev1.Pod, networkLi
 		}
 
 		// Secondary network information retrieved from API server. Proceed to configure secondary interface now.
-		if err = pc.configureSecondaryInterface(pod, network, resourceName, podCNIInfo, networkConfig); err != nil {
+		res, err := pc.configureSecondaryInterface(pod, network, resourceName, podCNIInfo, networkConfig)
+		if err != nil {
 			klog.ErrorS(err, "Secondary interface configuration failed",
 				"Pod", klog.KRef(pod.Namespace, pod.Name), "interface", network.InterfaceRequest,
 				"networkType", networkConfig.NetworkType)
 			savedErr = err
 		} else {
 			interfacesConfigured++
 		}
+		if res != nil {
+			status := &netdefv1.NetworkStatus{}
+			status.Name = network.Name
+			status.Default = false
+
+			for _, ifs := range res.Interfaces {
+				if ifs.Sandbox != "" {
+					status.Interface = ifs.Name
+					status.Mac = ifs.Mac
+				}
+			}
+
+			for _, ipconfig := range res.IPs {
+				status.IPs = append(status.IPs, ipconfig.Address.IP.String())
+			}
+
+			netStatus = append(netStatus, *status)
+		}
 	}
 
 	if savedErr != nil && interfacesConfigured == 0 {
 		// As we do not support secondary network update, do not return error to
 		// retry, if at least one secondary network is configured.
 		return savedErr
 	}
+
+	// Update the Pod's network status annotation
+	if netStatus != nil {
+		if err := netdefutils.SetNetworkStatus(pc.kubeClient, pod, netStatus); err != nil {
+			klog.ErrorS(err, "Pod network status annotation update failed", "Pod", klog.KRef(pod.Namespace, pod.Name))
+		} else {
+			klog.InfoS("Pod network status annotation updated successfully", "Pod", klog.KRef(pod.Namespace, pod.Name), "NetworkStatus", netStatus)
```

Comment:
Remove "successfully". Is the log generally useful? Would you consider a verbose level?